core/src/main/java/com/adobe/aem/commons/assetshare/components/details/impl/MetadataImpl.java

```diff
@@ -89,6 +99,9 @@ public class MetadataImpl extends AbstractEmptyTextComponent implements Metadata
     @ValueMapValue(name = MetadataImpl.PN_TYPE)
     private String typeString;
 
+    @ValueMapValue
+    private String jsonDataSourceProperty;
```

Comment:
Probably can just name this property `jsonSource` .. since its a property, its a little redundant to add `Property` to end of the property/var name